app/src/main/java/com/swent/mapin/ui/profile/ProfileViewModel.kt

```
@@ -96,33 +120,99 @@ class ProfileViewModel(
     private set
 
   init {
-    loadUserProfile()
+    try {
+      loadUserProfile()
+    } catch (e: Exception) {
+      // Gracefully handle initialization errors (e.g., Firebase not initialized in tests)
+      println("ProfileViewModel - Init error: ${e.message}")
+      _isLoading.value = false
+    }
   }
 
   /** Load user profile from Firestore. If profile doesn't exist, creates a default one. */
   fun loadUserProfile() {
     viewModelScope.launch {
-      _isLoading.value = true
-      val currentUser = FirebaseAuth.getInstance().currentUser
+      try {
+        _isLoading.value = true
+        val currentUser = FirebaseAuth.getInstance().currentUser
+
+        if (currentUser != null) {
+          // Try to load existing profile from Firestore
+          val existingProfile = repository.getUserProfile(currentUser.uid)
+
+          if (existingProfile != null) {
```

Remark (major): If the user isn't recognised i think it's a big issue and you shouldn't create a default user but rather throw an error and abort the action.

---

Yes, fix done.